odoo/addons/test_http/tests/test_webjson.py

```diff
@@ -129,51 +152,154 @@ def test_webjson_form_subtree(self):
             'records': stargates[:1]
         })
 
-        res_offset = self.url_open(f'{url}?offset=1')
-        res_offset.raise_for_status()
+        res_offset = self.url_open_json(f'{url}?offset=1')
         self.assertEqual(res_offset.json(), {
             'length': len(stargates),
             'records': stargates[1:]
         })
 
-        res_limit_offset = self.url_open(f'{url}?limit=1&offset=1')
-        res_limit_offset.raise_for_status()
+        res_limit_offset = self.url_open_json(f'{url}?limit=1&offset=1')
         self.assertEqual(res_limit_offset.json(), {
             'length': len(stargates),
             'records': stargates[1:2]
         })
+
+    def test_webjson_list_domain(self):
+        env = self.authenticate_demo()
+        domain = [("address", "like", "gs38")]
+        res = self.url_open_json(f'/test_http.stargate?domain={domain}')
+        self.assertEqual(
+            res.json(),
+            env['test_http.stargate']
+                .web_search_read(domain, {'name': {}, 'sgc_designation': {}})
+        )
+
+    def test_webjson_list_args(self):
+        env = self.authenticate_demo()
+        # create a default filter
+        domain = [("name", "ilike", "earth")]
+        self.env['ir.filters'].create({
+            'name': 'my filter',
+            'is_default': True,
+            'domain': domain,
+            'model_id': 'test_http.stargate',
+        })
+        res = self.url_open_json('/test_http.stargate')
+        self.assertEqual(
+            res.json(),
+            env['test_http.stargate']
+                .web_search_read(domain, {'name': {}, 'sgc_designation': {}})
+        )
+        # we should find one redirect with the domain in the URL
+        [hist] = res.history
+        self.assertEqual(hist.status_code, 307)
+        str_domain = str(domain).replace(' ', '+')
+        self.assertIn("limit=80", res.url)
+        self.assertIn(f"domain={str_domain}", res.url)
+
+    def test_webjson_pivot(self):
+        env = self.authenticate_demo()
+        res = self.url_open_json('/test_http.stargate?view_type=pivot')
+        self.assertEqual(
+            res.json(),
+            self._read_group_list(
+                env['test_http.stargate'], [], ['galaxy_id', 'has_galaxy_crystal'], ['availability']),
+        )
+
+        res = self.url_open_json('/test_http.stargate?view_type=pivot&groupby=has_galaxy_crystal&fields=availability:min')
+        self.assertEqual(
+            res.json(),
+            self._read_group_list(env['test_http.stargate'], [], ['has_galaxy_crystal'], ['availability:min']),
+        )
+
+    def test_webjson_graph(self):
+        env = self.authenticate_demo()
+        res = self.url_open_json('/test_http.stargate?view_type=graph')
+        self.assertEqual(
+            res.json(),
+            self._read_group_list(env['test_http.stargate'], [], ['galaxy_id']),
+        )
+
+    def test_webjson_crm_default_filter(self):
+        self.action_crm()
```

Comment:
Adding a little test for crm was a quick and dirty hack to make sure loading crm would work. Let's not use crm as test data but instead define our own views inside test_http